aim/sdk/objects/image.py

```diff
@@ -102,9 +101,76 @@ def json(self):
             'height': self.height,
         }
 
+    def _from_pil_image(self, pil_image: PILImage.Image):
+        assert isinstance(pil_image, PILImage.Image)
+        image_bytes_p = BytesIO()
+        pil_image.save(image_bytes_p, format='png')
 
-def convert_to_aim_image(obj):
-    if isinstance(obj, PILImage):
-        return Image.from_pil_image(obj)
-    # TODO support other sources as well
-    raise ValueError
+        self.storage['data'] = BLOB(data=image_bytes_p.getvalue())
+        self.storage['source'] = 'PIL.Image'
+        self.storage['mode'] = pil_image.mode
+        self.storage['format'] = 'png'
+        self.storage['width'], self.storage['height'] = pil_image.size
+
+    def _from_numpy_array(self, array: np.ndarray):
+        if array.ndim not in {2, 3}:
```

Comment:
what about rgba mode? ndim=4
doesn't ndim of grayscale mode equal to 1?

Reply:
@gorarakelyan `ndim` is still 3 in this case, the value of third dimension would be 4